multivac/gather_test_data.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+import argparse
+import glob
+import json
+import os
+from datetime import datetime
+
+from gather_job_data import github_time_to_unix
+from influxdb import influx_connector
+from sensors.test_status import test_status_iter
+
+
+class GatherTestData:
+    def __init__(self, cli_args):
+        self.workflow_run_jobs_dir = 'workflow_run_jobs'
+        self.job_log_files = sorted(glob.glob(
+            os.path.join(self.workflow_run_jobs_dir, '*[0-9].log')),
+            reverse=True)
+        self.output_dir = 'output'
+        self.gathered_test_data = {}
+        self.since_seconds = None
+
+        since: str = cli_args.since
+        if since and len(since) > 1:
+            wrong_usage_message = 'Wrong \'--since\' option: got wrong {}. ' \
+                                  'Usage: NN[d|h], example: 42d.'
+            if since.endswith('d'):
+                try:
+                    self.since_seconds = int(since.rstrip('d')) * 86400
+                except ValueError:
+                    print(wrong_usage_message.format('number'))
+            elif since.endswith('h'):
+                try:
+                    self.since_seconds = int(since.rstrip('h')) * 3600
+                except ValueError:
+                    print(wrong_usage_message.format('number'))
+            else:
+                print(wrong_usage_message.format('unit'))
+
+    def collect_data(self):
+        curr_time = datetime.timestamp(datetime.now())
+        print('Starting to collect data...')
+        for log in self.job_log_files:
+            log_id = log.lstrip(self.workflow_run_jobs_dir + '/').rstrip('.log')
+            self.gathered_test_data[log_id] = list()
+            with open(log, 'r') as log_file:
+                log_file_list = list(log_file)
+            test_number = 1
+
+            if self.since_seconds:
+                time = github_time_to_unix(log_file_list[0].split(
+                    ' ', 1)[0].split('.', 1)[0])
+                if curr_time - time > self.since_seconds:
+                    self.gathered_test_data.pop(log_id)
+                    break
+                    # Because list of log files is sorted
```

Comment:
good, but can be made more clear:

```suggestion
                    # List of log files is sorted by time: latest to earliest.
                    # So when we find the first log which is out of selected time limit,
                    # all further logs are also out of limit.
```

Reply:
tnx, done